stubs/docker/docker/_types.pyi

```diff
@@ -1,8 +1,18 @@
 # Internal-use module for types shared by multiple modules.
 # This does not match a module in docker-py.
 
-from typing_extensions import TypeAlias
+from typing import TypedDict, type_check_only
+from typing_extensions import NotRequired, TypeAlias
 
 # Type alias for JSON, explained at:
 # https://github.com/python/typing/issues/182#issuecomment-1320974824.
 JSON: TypeAlias = dict[str, JSON] | list[JSON] | str | int | float | bool | None
+
+# See https://docs.docker.com/engine/api/v1.42/#tag/Container/operation/ContainerWait
+@type_check_only
+class _WaitErrorDetails(TypedDict):
+    Message: str
+
+class WaitContainerResponse(TypedDict):
```

Comment:
We should handle this consistently. I.e. add `@type_check_only` to `WaitContainerResponse` and add the underscore to the name to signal this even when not looking at the documentation.

Reply:
@srittau I added `@type_check_only` as that was a miss.

My thinking is that we should not add an underscore. To me, the underscore in `_types.pyi` says "this is internal" and the underscore in `_WaitErrorDetails` says "this is private to `_types.pyi`. Having no underscore in `WaitContainerResponse` says that `WaitContainerResponse` is part of the interface of `_types.pyi`. Similarly, a private class does not need all of its methods to be private.

Please let me know what you think, given the above, and I'm happy to make the change if you still prefer an underscore.

Reply:
When trying to add the underscore, the `flake8` pre-commit gives an error:

```
stubs/docker/docker/_types.pyi:17:1: Y049 TypedDict "_WaitContainerResponse" is not used
```

```diff
diff --git a/stubs/docker/docker/_types.pyi b/stubs/docker/docker/_types.pyi
index 0a6e03c94..97b9a9247 100644
--- a/stubs/docker/docker/_types.pyi
+++ b/stubs/docker/docker/_types.pyi
@@ -14,6 +14,6 @@ class _WaitErrorDetails(TypedDict):
     Message: str
 
 @type_check_only
-class WaitContainerResponse(TypedDict):
+class _WaitContainerResponse(TypedDict):
     StatusCode: int
     Error: NotRequired[_WaitErrorDetails]
diff --git a/stubs/docker/docker/api/container.pyi b/stubs/docker/docker/api/container.pyi
index b1e025fe6..88b943fe1 100644
--- a/stubs/docker/docker/api/container.pyi
+++ b/stubs/docker/docker/api/container.pyi
@@ -3,7 +3,7 @@ from _typeshed import Incomplete
 from typing import Literal, TypedDict, overload, type_check_only
 from typing_extensions import TypeAlias
 
-from docker._types import WaitContainerResponse
+from docker._types import _WaitContainerResponse
 from docker.types.daemon import CancellableStream
 
 from ..types import ContainerConfig, EndpointConfig, HostConfig, NetworkingConfig
@@ -165,4 +165,4 @@ class ContainerApiMixin:
         container: _Container,
         timeout: int | None = None,
         condition: Literal["not-running", "next-exit", "removed"] | None = None,
-    ) -> WaitContainerResponse: ...
+    ) -> _WaitContainerResponse: ...
diff --git a/stubs/docker/docker/models/containers.pyi b/stubs/docker/docker/models/containers.pyi
index 3cdb5ca1e..ca8c64917 100644
--- a/stubs/docker/docker/models/containers.pyi
+++ b/stubs/docker/docker/models/containers.pyi
@@ -2,7 +2,7 @@ import datetime
 from _typeshed import Incomplete
 from typing import Literal, NamedTuple, overload
 
-from docker._types import WaitContainerResponse
+from docker._types import _WaitContainerResponse
 from docker.types.daemon import CancellableStream
 
 from .images import Image
@@ -100,7 +100,7 @@ class Container(Model):
     ): ...
     def wait(
         self, *, timeout: float | None = None, condition: Literal["not-running", "next-exit", "removed"] | None = None
-    ) -> WaitContainerResponse: ...
+    ) -> _WaitContainerResponse: ...
 
 class ContainerCollection(Collection[Container]):
     model: type[Container]
```